task_sdk/src/airflow/sdk/api/datamodels/_generated.py

```diff
@@ -239,7 +239,7 @@ class TaskInstance(BaseModel):
     dag_id: Annotated[str, Field(title="Dag Id")]
     run_id: Annotated[str, Field(title="Run Id")]
     try_number: Annotated[int, Field(title="Try Number")]
-    map_index: Annotated[int, Field(title="Map Index")] = -1
+    map_index: Annotated[int | None, Field(title="Map Index")] = None
```

Comment:
Same question, when can this be None?

Reply:
Currently it can't, but the `map_index=-1`  was only to because map_index was part of the primary key of the row, and you can't have a null value in a PK column.

This change is to support that change (so that the SDK doesn't have to know that -1 = non mapped, that can be kept entirely in the server side to eventually in 3.1 or 3.2 be silently changed.)

Reply:
So the goal is somewhat like this?

* In SDK side, None means unmapped and (non-negative) int means mapped
* In DB, -1 means unmapped and non-negative int means mapped

I wonder if we should also add something to ensure negative ints are eliminated when None is used.

Reply:
This is working _towards_ having None in the DB now too. We can do that now that the PK of task_instance does not contain map_index column (it is just `(id)` now)

Reply:
Maybe I should remove this change here, it's not needed for this and is likely confusing

Reply:
That would do